app/models/team.rb

```diff
@@ -63,10 +63,21 @@ def in_current_season
       where(season: Season.current)
     end
 
+    def by_season_year(year)
+      where(season: Season.where(name: year))
+    end
+
+    def select_teams_by_season_year(year)
+      by_season_year(year).selected
+    end
+
     def selected
       where(kind: %w(sponsored voluntary))
     end
 
+    def activities_ordered_by(direction)
```

Comment:
This, too, seems like a highly specialised method and to me it doesn't feel like ordering the activities is part of the `Team` class' responsibility. I would move it back into controller.